src/plugins/Polls/UI/PollsDialog.tsx

```diff
@@ -0,0 +1,348 @@
+    const [polls, setPolls] = useState<Array<PollGunDB>>([])
+    const classes = useStylesExtends(useNewPollStyles(), props)
+
+    useEffect(() => {
+        Services.Plugin.invokePlugin('maskbook.polls', 'getExistingPolls').then((polls) => {
+            const myPolls: Array<PollGunDB> = []
+            polls.map((poll) => {
+                if (poll.id === props.senderId) {
+                    myPolls.push(poll)
+                }
+            })
+            setPolls(polls.reverse())
+        })
+    }, [props.senderId])
+
+    const insertPoll = (poll?: PollMetaData | null) => {
+        props.onSelectExistingPoll(poll)
+    }
+
+    return (
+        <div className={classes.wrapper}>
+            {polls.map((p) => (
+                <PollCardUI onClick={() => insertPoll(p)} poll={p} key={p.key as string | number} />
+            ))}
+        </div>
+    )
+}
+
+const useStyles = makeStyles((theme) => {
+    createStyles({
+        title: {
+            marginLeft: 6,
+        },
+        container: {
+            width: '100%',
+        },
+    })
+})
+
+interface PollsDialogProps
+    extends withClasses<
+        | KeysInferFromUseStyles<typeof useStyles>
+        | 'dialog'
+        | 'wrapper'
+        | 'backdrop'
+        | 'container'
+        | 'close'
+        | 'header'
+        | 'content'
+        | 'paper'
+        | 'title'
+        | 'label'
+        | 'button'
+    > {
+    open: boolean
+    onConfirm: (opt?: any) => void
+    onDecline: () => void
+    DialogProps?: Partial<DialogProps>
+}
+
+export default function PollsDialog(props: PollsDialogProps) {
+    const classes = useStylesExtends(useStyles(), props)
+    const state = useState(0)
+    const [, setTabState] = state
+    const loading = useState(false)
+
+    const createNewPoll = () => {
+        setTabState(1)
+    }
+
+    const insertPoll = (data?: PollMetaData | null) => {
+        const ref = getActivatedUI().typedMessageMetadata
+        const next = new Map(ref.value.entries())
+        data ? next.set('poll', data) : next.delete('poll')
+        ref.value = next
+        props.onConfirm()
+    }
+
+    const tabProps: AbstractTabProps = {
+        tabs: [
+            {
+                label: 'Create New',
+                children: (
+                    <NewPollUI
+                        {...props}
+                        loading={loading}
+                        createNewPoll={createNewPoll}
+                        senderName={useCurrentIdentity()?.linkedPersona?.nickname}
+                        senderId={useCurrentIdentity()?.linkedPersona?.fingerprint}
+                    />
+                ),
+                p: 0,
+            },
+            {
+                label: 'Select Existing',
+                children: (
+                    <ExistingPollsUI
+                        {...props}
+                        onSelectExistingPoll={insertPoll}
+                        senderId={useCurrentIdentity()?.linkedPersona?.fingerprint}
```

Comment:
Please extract those hooks calls to a variable, React hooks is not free (in the meaning of computation resources) to use.